Source/OxyPlot/Series/DataPointHelper.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+
+namespace OxyPlot.Series
+{
+	/// <summary>
+	/// Data point helper functions.
+	/// </summary>
+	public class DataPointHelper
+	{
+		/// <summary>
+		/// Find index of max(x) &lt;= target x in a list of data points
+		/// </summary>
+		/// <param name='items'>
+		/// vector of data points
+		/// </param>
+		/// <param name='targetX'>
+		/// target x.
+		/// </param>
+		/// <param name='guess'>
+		/// initial guess index.
+		/// </param>
+		/// <returns>
+		/// index of x with max(x) &lt;= target x or -1 if cannot find
+		/// </returns>
+		public static int FindIndex(List<DataPoint> items, double targetX, int guess)
```

Comment:
I am a bit reluctant to adding such static methods to the library. But I see it can be used in two series types, then it might be ok...